### PR TITLE
Use let extension to remove bang operator from nullable variables

### PR DESCRIPTION
**Pull request by bitfriend**
This small PR was split from [bigger PR](https://github.com/acterglobal/a3/pull/2125) so that reviewer can see changes easily.

### CODE REVIEW

File: app/lib/common/toolkit/errors/inline_error_button.dart
Comment (gnunicorn) [inline app/lib/common/toolkit/errors/inline_error_button.dart:49]: this is incorrect though. if we don't have an `onRetryTap` we shouldn't hand in an empty function but `null`.

```suggestion
            onRetryTap: onRetryTap.let((cb) => () {
              cb();
              Navigator.pop(context);
            }),
```

File: app/lib/common/toolkit/errors/inline_error_button.dart
Comment (gnunicorn) [inline app/lib/common/toolkit/errors/inline_error_button.dart:78]: same as above.

File: app/lib/common/toolkit/errors/inline_error_button.dart
Comment (gnunicorn) [inline app/lib/common/toolkit/errors/inline_error_button.dart:70]: we can spell it out: `Icon icon`. Generally in dart or rust abbreviating short words isn't really that great.

File: app/lib/common/utils/utils.dart
Comment (gnunicorn) [inline app/lib/common/utils/utils.dart:377]: I don't feel like using `p0` is a very nice way of describing these. Let's stick to using proper words that make it easier to follow along later:
```suggestion
  user.firstName.let((name) {
    if (name.isNotEmpty) initials += name[0].toUpperCase();
  });
  user.lastName.let((name) {
    if (name.isNotEmpty) initials += name[0].toLowerCase();
  });
```

File: app/lib/common/widgets/acter_icon_picker/acter_icon_widget.dart
Comment (gnunicorn) [inline app/lib/common/widgets/acter_icon_picker/acter_icon_widget.dart:31]: this is wrong. this should happen in `initState` and `didUpdateDependencies` (pseudocode):

```
  void _setWidgetValues() {
    widget.color.let((p0) => color.value = p0);
    widget.icon.let((p0) => icon.value = p0);
  }

   @override
   void initState() {
     _setWidgetValues()
   }

   @override
   void didUpdateDependencies() {
     _setWidgetValues()
   }
```

Comment (gnunicorn) [reply app/lib/common/widgets/acter_icon_picker/acter_icon_widget.dart:31]: I think this is maybe why we sometimes see the color not being properly reflected in the UI despite selecting an icon.

File: app/lib/common/widgets/acter_icon_picker/acter_icon_widget.dart
Comment (gnunicorn) [inline app/lib/common/widgets/acter_icon_picker/acter_icon_widget.dart:35]: same as above: if we don't have a callback we want there to be `null` not an empty function - as the latter behaves differently in inkwell.

```suggestion
      onTap: widget.onIconSelection.let(
        (cb) => () => showActerIconPicker(
```

**Review by gnunicorn — APPROVED** [review]
Nice work.

I have a few things I'd like to see changed, and once done you can merge it (without another review).

File: app/lib/common/toolkit/errors/inline_error_button.dart
Comment (bitfriend) [reply app/lib/common/toolkit/errors/inline_error_button.dart:70]: Frankly, I renamed it from `icon` to `icn` in order to avoid confusing with `this.icon`.

File: app/lib/common/utils/utils.dart
Comment (bitfriend) [reply app/lib/common/utils/utils.dart:377]: Yeah, I just wanted to use the word that has universal meaning, so that other developers can copy/paste and use this code instantly.

Comment (bitfriend) [reply app/lib/common/utils/utils.dart:377]: @gnunicorn Do you want to use specialized word not universal word?

Comment (gnunicorn) [reply app/lib/common/utils/utils.dart:377]: I prefer explicit words, yes.

File: app/lib/common/toolkit/errors/inline_error_button.dart
Comment (gnunicorn) [reply app/lib/common/toolkit/errors/inline_error_button.dart:70]: well, okay then.

File: app/lib/common/toolkit/errors/inline_error_button.dart
Comment (bitfriend) [reply app/lib/common/toolkit/errors/inline_error_button.dart]: Fixed by https://github.com/acterglobal/a3/pull/2233/commits/313bf8d1392ea01c0add35d03a523613e590136f

File: app/lib/common/toolkit/errors/inline_error_button.dart
Comment (bitfriend) [reply app/lib/common/toolkit/errors/inline_error_button.dart]: Fixed by https://github.com/acterglobal/a3/pull/2233/commits/313bf8d1392ea01c0add35d03a523613e590136f

File: app/lib/common/widgets/acter_icon_picker/acter_icon_widget.dart
Comment (bitfriend) [reply app/lib/common/widgets/acter_icon_picker/acter_icon_widget.dart]: Fixed by https://github.com/acterglobal/a3/pull/2233/commits/313bf8d1392ea01c0add35d03a523613e590136f

File: app/lib/common/widgets/acter_icon_picker/acter_icon_widget.dart
Comment (bitfriend) [reply app/lib/common/widgets/acter_icon_picker/acter_icon_widget.dart]: Fixed by https://github.com/acterglobal/a3/pull/2233/commits/cbe3ef9e87c69d98ee2744633224b93dbb461c8d

File: app/lib/common/utils/utils.dart
Comment (bitfriend) [reply app/lib/common/utils/utils.dart:377]: Frankly, I wanted to use `p0` for object and `cb` for callback.
Don't worry.
Will fix.

Comment (bitfriend) [reply app/lib/common/utils/utils.dart:377]: Fixed by https://github.com/acterglobal/a3/pull/2233/commits/68d5e20f88fd02b863a771d7b583630301a41158

Comment (gnunicorn) [reply app/lib/common/utils/utils.dart:377]: cb for callback is specific enough.

**Pull request merged**
